bson.c

```diff
@@ -1212,7 +1260,7 @@ char* bson_to_zval(char *buf, HashTable *result, mongo_bson_conversion_options *
 
 				/* put it all together */
 				array_init(value);
-				add_assoc_stringl(value, "$ref", ns, ns_len-1, 1);
+				add_assoc_zval(value, "$ref", zns);
```

Comment:
AFAIK, casting a `const char *` to a `char *` would be the same warning as passing the original pointer to `add_assoc_stringl_ex()` -- it downgrades the pointer type.